c/main.c

```diff
@@ -0,0 +1,150 @@
+#include <deltachat.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+void handle_message(dc_context_t *context, int chat_id, int message_id) {
+  dc_chat_t *chat = dc_get_chat(context, chat_id);
+
+  if (dc_chat_get_type(chat) == DC_CHAT_TYPE_SINGLE) {
+    dc_msg_t *msg = dc_get_msg(context, message_id);
+
+    if (!msg) {
+      return;
+    }
+
+    dc_send_text_msg(context, chat_id, dc_msg_get_text(msg));
+    dc_msg_unref(msg);
+  }
+
+  dc_chat_unref(chat);
+}
+
+void *event_handler(void *context) {
+  dc_event_emitter_t *emitter = dc_get_event_emitter(context);
+  dc_event_t *event;
+  while ((event = dc_get_next_event(emitter)) != NULL) {
+    // use the event as needed, e.g. dc_event_get_id() returns the type.
+    // once you're done, unref the event to avoid memory leakage:
+    int event_type = dc_event_get_id(event);
+
+    if (event_type == DC_EVENT_ERROR || event_type == DC_EVENT_INFO ||
```

Comment:
Could be a `switch..case`?

Reply:
might be annoying with the break statements, but good in this specific case

Reply:
doing it as a switch statement gives me errors:
```
main.c: In function ‘event_handler’:
main.c:37:7: error: a label can only be part of a statement and a declaration is not a statement
   37 |       char *message = dc_event_get_data2_str(event);
      |       ^~~~
main.c:53:7: error: a label can only be part of a statement and a declaration is not a statement
   53 |       int progress = dc_event_get_data1_int(event);
      |       ^~~
main.c:54:7: error: expected expression before ‘char’
   54 |       char *comment = dc_event_get_data2_str(event);
      |       ^~~~
main.c:55:56: error: ‘comment’ undeclared (first use in this function)
   55 |       printf("[configure-progress] %d %s\n", progress, comment);
      |                                                        ^~~~~~~
main.c:55:56: note: each undeclared identifier is reported only once for each function it appears in
main.c:69:7: error: a label can only be part of a statement and a declaration is not a statement
   69 |       int chat_id = dc_event_get_data1_int(event);
      |       ^~~
main.c:70:7: error: expected expression before ‘int’
   70 |       int message_id = dc_event_get_data2_int(event);
      |       ^~~
main.c:71:48: error: ‘message_id’ undeclared (first use in this function); did you mean ‘message’?
   71 |       printf("[msg-changed] %d %d\n", chat_id, message_id);
      |                                                ^~~~~~~~~~
      |                                                message
main.c:85:7: error: a label can only be part of a statement and a declaration is not a statement
   85 |       int chat_id = dc_event_get_data1_int(event);
      |       ^~~
main.c:85:11: error: redefinition of ‘chat_id’
   85 |       int chat_id = dc_event_get_data1_int(event);
      |           ^~~~~~~
main.c:69:11: note: previous definition of ‘chat_id’ was here
   69 |       int chat_id = dc_event_get_data1_int(event);
      |           ^~~~~~~
main.c:86:7: error: expected expression before ‘int’
   86 |       int message_id = dc_event_get_data2_int(event);
      |       ^~~
```
```diff
diff --git a/c/main.c b/c/main.c
index 55b7661..d197b94 100644
--- a/c/main.c
+++ b/c/main.c
@@ -30,8 +30,10 @@ void *event_handler(void *context) {
     // once you're done, unref the event to avoid memory leakage:
     int event_type = dc_event_get_id(event);
 
-    if (event_type == DC_EVENT_ERROR || event_type == DC_EVENT_INFO ||
-        event_type == DC_EVENT_WARNING) {
+    switch (event_type) {
+    case DC_EVENT_ERROR:
+    case DC_EVENT_INFO:
+    case DC_EVENT_WARNING:
       char *message = dc_event_get_data2_str(event);
 
       switch (event_type) {
@@ -46,8 +48,8 @@ void *event_handler(void *context) {
         break;
       }
       dc_str_unref(message);
-    } else if (event_type == DC_EVENT_CONFIGURE_PROGRESS) {
-
+      break;
+    case DC_EVENT_CONFIGURE_PROGRESS:
       int progress = dc_event_get_data1_int(event);
       char *comment = dc_event_get_data2_str(event);
       printf("[configure-progress] %d %s\n", progress, comment);
@@ -62,8 +64,8 @@ void *event_handler(void *context) {
         dc_start_io(context);
       }
       dc_str_unref(comment);
-
-    } else if (event_type == DC_EVENT_MSGS_CHANGED) {
+      break;
+    case DC_EVENT_MSGS_CHANGED:
       int chat_id = dc_event_get_data1_int(event);
       int message_id = dc_event_get_data2_int(event);
       printf("[msg-changed] %d %d\n", chat_id, message_id);
@@ -78,13 +80,16 @@ void *event_handler(void *context) {
         handle_message(context, new_chat_id, message_id);
       }
       dc_msg_unref(msg);
-    } else if (event_type == DC_EVENT_INCOMING_MSG) {
+      break;
+    case DC_EVENT_INCOMING_MSG:
       int chat_id = dc_event_get_data1_int(event);
       int message_id = dc_event_get_data2_int(event);
       printf("[incoming-msg] %d %d\n", chat_id, message_id);
       handle_message(context, chat_id, message_id);
-    } else {
+      break;
+    default:
       printf("[?] unhandled event of type: %d\n", event_type);
+      break;
     }
 
     dc_event_unref(event);
```

Reply:
keeping the if statement for now